xarray/core/indexes.py

```diff
@@ -1,12 +1,200 @@
 import collections.abc
-from typing import Any, Dict, Hashable, Iterable, Mapping, Optional, Tuple, Union
+from contextlib import suppress
+from datetime import timedelta
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Hashable,
+    Iterable,
+    Mapping,
+    Optional,
+    Tuple,
+    Union,
+)
 
 import numpy as np
 import pandas as pd
 
-from . import formatting
+from . import formatting, utils
+from .indexing import ExplicitlyIndexedNDArrayMixin, NumpyIndexingAdapter
+from .npcompat import DTypeLike
 from .utils import is_scalar
-from .variable import Variable
+
+if TYPE_CHECKING:
+    from .variable import Variable
+
+
+class Index:
+    """Base class inherited by all xarray-compatible indexes."""
+
+    __slots__ = ("coord_names",)
+
+    def __init__(self, coord_names: Union[Hashable, Iterable[Hashable]]):
+        if isinstance(coord_names, Iterable) and not isinstance(coord_names, str):
+            self.coord_names = tuple(coord_names)
+        else:
+            self.coord_names = tuple([coord_names])
```

Comment:
```suggestion
        if isinstance(coord_names, str):
            coord_names = (coords_names,)
        self.coord_names = tuple(coords_names)
```

This is how we do normalization of the `dims` argument in `xarray.Variable`. Unless you can think of a good reason to do otherwise, I would suggest copying it here.